flamingpy/codes/graphs/stabilizer_graph.py

```diff
@@ -289,20 +297,47 @@ def real_nodes(self):
     def real_edges(self):
         """Returns an iterable of all edges excluding the ones connected to the
         'low' or 'high' points."""
-        is_high_or_low = lambda n: n in ("low", "high")
+
+        def is_high_or_low(n):
+            return n in ("low", "high")
+
         return filter(
             lambda edge: not (is_high_or_low(edge[0]) or is_high_or_low(edge[1])),
             self.edges(),
         )
 
+    def assign_weights(self, code):
+        """Assign the weights to the graph based on the weight
+        of the common vertex of each stabilizer pair of the code."""
+        for edge in self.edges():
+            data = self.edge_data(*edge)
+            if data["common_vertex"] is not None:
+                data["weight"] = code.graph.nodes[data["common_vertex"]]["weight"]
+            else:
+                data["weight"] = 0.0
+
+    def to_nx(self):
+        """Convert the same graph into a NxStabilizerGraph.
+
+        This involves converting the retworkx graph representation to a
+        networkx graph representation.
+        """
+        if isinstance(self, NxStabilizerGraph):
+            return self
+        nx_graph = NxStabilizerGraph()
+        for edge in self.edges():
+            nx_graph.add_edge(*edge, self.edge_data(*edge)["common_vertex"])
+            nx_graph.edge_data(*edge)["weight"] = self.edge_data(*edge)["weight"]
+        return nx_graph
+
     def draw(self, **kwargs):
         """Draw the stabilizer graph with matplotlib.
 
         See flamingpy.utils.viz.draw_dec_graph for more details.
         """
         from flamingpy.utils.viz import draw_dec_graph
 
-        draw_dec_graph(self, **kwargs)
+        draw_dec_graph(self.to_nx(), **kwargs)
```

Comment:
If you agree with my previous comment, this might need to move to NxStabilizerGraph.

Reply:
I don't. 😄